src/tests/components/forms/FamilyForm.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { configure, screen } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { render, cleanup } from '@testing-library/react'
+import {
+  configMock,
+  mockDocument,
+  mockFamiliesData,
+} from '@/tests/utilsTest/mocks'
+import { FamilyForm } from '@/components/forms/FamilyForm'
+import { TFamily } from '@/interfaces'
+import 'jest-localstorage-mock'
+import { TestWrapper } from '@/tests/utilsTest/render'
+
+const flushPromises = async () => new Promise(process.nextTick)
+
+// useBlocker only can be used in a router context
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useBlocker: jest.fn(),
+}))
+
+// API calls
+jest.mock('axios', () => {
+  return {
+    create: jest.fn(() => ({
+      get: jest.fn(),
+      defaults: {
+        headers: {
+          common: {},
+        },
+      },
+    })),
+  }
+})
+
+jest.mock('@/api', () => ({
+  getApiUrl: jest.fn().mockReturnValue('http://mock-api-url'),
+}))
+
+const localStorageMock = {
+  getItem: (_: string) => 'token',
+  setItem: jest.fn(),
+  clear: jest.fn(),
+  length: 1,
+  key: jest.fn(),
+  removeItem: jest.fn(),
+}
+
+global.localStorage = localStorageMock
+
+jest.mock('@/hooks/useConfig', () =>
```

Comment:
Just so that I understand fully, what we're saying is that each component needs to contain it's own mock rather than how it was previous?